cvat/apps/dataset_manager/formats/utils.py

```diff
@@ -61,19 +62,37 @@ def make_colormap(instance_data):
 
     return {label['name']: [hex2rgb(label['color']), [], []] for label in labels}
 
+def generate_color(color, used_colors):
+    def tint_shade_color():
+        for added_color in (255, 0):
+            for factor in range(1, 10):
+                yield tuple(map(lambda c: int(c + (added_color - c) * factor / 10), color))
 
-def get_label_color(label_name, label_names):
+    def get_unused_color():
+        sorted_colors = sorted(used_colors)
+        max_dist = max(zip(sorted_colors, sorted_colors[1:]), key=lambda c_pair: c_pair[1][0] - c_pair[0][0])
+
+        return ((max_dist[0][0] + max_dist[1][0]) // 2, max_dist[0][1], max_dist[0][2])
```

Comment:
Hi, @azhavoro, Is it the final version of this patch? If it is, then I have a question:
Will this solution work correctly if all used colors have a same R-value? For example in case:
```
used_colors = [(100, 0, 0), (100, 50, 0), (100, 0, 50)]
```
this code will return `(100, 0, 0)` but this color is already use. I'm not sure that such situation is very likely, but it seems to me that we can extend this algorithm to other components of the color as well.

Reply:
Agree, fixed.